src/core/client/admin/routes/login/containers/AccountCompletionContainer.tsx

```diff
@@ -0,0 +1,141 @@
+import { withRouter, WithRouter } from "found";
+import * as React from "react";
+import { Component } from "react";
+
+import { AccountCompletionContainer_auth as AuthData } from "talk-admin/__generated__/AccountCompletionContainer_auth.graphql";
+import { AccountCompletionContainer_me as UserData } from "talk-admin/__generated__/AccountCompletionContainer_me.graphql";
+import { AccountCompletionContainerLocal as Local } from "talk-admin/__generated__/AccountCompletionContainerLocal.graphql";
+import {
+  CompleteAccountMutation,
+  SetAuthViewMutation,
+  SetRedirectPathMutation,
+  withCompleteAccountMutation,
+  withSetAuthViewMutation,
+  withSetRedirectPathMutation,
+} from "talk-admin/mutations";
+import {
+  graphql,
+  withFragmentContainer,
+  withLocalStateContainer,
+} from "talk-framework/lib/relay";
+
+type Props = {
+  completeAccount: CompleteAccountMutation;
+  setAuthView: SetAuthViewMutation;
+  local: Local;
+  auth: AuthData;
+  me: UserData | null;
+  setRedirectPath: SetRedirectPathMutation;
+} & WithRouter;
+
+function handleAccountCompletion(props: Props) {
+  const {
+    local: { authView },
+    me,
+    auth,
+    setAuthView,
+  } = props;
+  if (me) {
+    if (!me.email) {
+      if (authView !== "ADD_EMAIL_ADDRESS") {
+        setAuthView({ view: "ADD_EMAIL_ADDRESS" });
```

Comment:
The logic here makes sense, but this function has a high cyclomatic complexity, further aided by the fact that it doesn't return early. I'd suggest returning early if the `authView` is set.